src/tables.jl

```diff
@@ -126,25 +126,37 @@ uniquely identify rows in the `table` then an `ArgumentError` is throw. If `forc
 then the duplicate (non-unique) entries will be overwritten.
 """
 function populate!(A, table, value::Symbol; force=false)
-    # Use a BitArray mask to detect duplicates and error instead of overwriting.
-    mask = force ? falses() : falses(size(A))
-
-    for r in Tables.rows(table)
-        vals = Tuple(Tables.getcolumn(r, c) for c in dimnames(A))
-        inds = map(findindex, vals, axiskeys(A))
-
-        # Handle duplicate error checking if applicable
-        if !force
-            # Error if mask already set.
-            mask[inds...] && throw(ArgumentError("Key $vals is not unique"))
-            # Set mask, marking that we've set this index
-            setindex!(mask, true, inds...)
-        end
+    cols = Tables.columns(table)
 
-        # Insert our value into the data array
-        setindex!(A, Tables.getcolumn(r, value), inds...)
+    indices = CartesianIndex.(
+        map(dimnames(A)) do nm
+            col = Tables.getcolumn(cols, nm)
+            inds = axiskeys(A, nm)
+            result = zeros(Int, length(col))
+
+            for idx in pairs(inds)
+                result[last(idx) .== col] .= first(idx)
```

Comment:
The broadcasted `.==` index seems to be faster and more memory efficient than `findall` or looping ourselves.